h2o-algos/src/main/java/hex/gam/GamSplines/ThinPlateDistanceWithKnots.java

```diff
@@ -0,0 +1,105 @@
+package hex.gam.GamSplines;
+
+import hex.DataInfo;
+import hex.glm.GLMModel.GLMParameters.MissingValuesHandling;
+import hex.util.LinearAlgebraUtils.BMulInPlaceTask;
+import water.MRTask;
+import water.MemoryManager;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.NewChunk;
+import water.fvec.Vec;
+
+import static hex.gam.GAMModel.GAMParameters;
+import static hex.gam.GamSplines.ThinPlateRegressionUtils.*;
+import static hex.genmodel.algos.gam.GamUtilsThinPlateRegression.calculateDistance;
+import static org.apache.commons.math3.util.CombinatoricsUtils.factorial;
+import static water.util.ArrayUtils.transpose;
+
+// Implementation details of this class can be found in GamThinPlateRegressionH2O.doc attached to this 
```

Comment:
:+1: for the link. But please, use java docstring for classes.

Reply:
Done.